src/main/java/ui/components/pickers/LabelPickerUILogic.java

```diff
@@ -1,382 +1,27 @@
-
-    private void addExistingLabels() {
-        // used once to populate topLabels at the start
-        allLabels.stream()
-                .filter(label -> issue.getLabels().contains(label.getActualName()))
-                .forEach(label -> topLabels.add(new PickerLabel(label, this, true)));
-    }
-
-    private void preProcessAndUpdateTopLabels(String name) {
-        Optional<TurboLabel> turboLabel =
-                allLabels.stream().filter(label -> label.getActualName().equals(name)).findFirst();
-        if (turboLabel.isPresent()) {
-            if (turboLabel.get().isExclusive() && !resultList.get(name)) {
-                // exclusive label check
-                String group = turboLabel.get().getGroup().get();
-                allLabels
-                        .stream()
-                        .filter(TurboLabel::isExclusive)
-                        .filter(label -> label.getGroup().get().equals(group))
-                        .forEach(label -> updateTopLabels(label.getActualName(), false));
-                updateTopLabels(name, true);
-            } else {
-                updateTopLabels(name, !resultList.get(name));
-            }
-        }
-    }
-
-    private void updateTopLabels(String name, boolean isAdd) {
-        // adds new labels to the end of the list
-        resultList.put(name, isAdd); // update resultList first
-        if (isAdd) {
-            if (issue.getLabels().contains(name)) {
-                topLabels.stream()
-                        .filter(label -> label.getActualName().equals(name))
-                        .forEach(label -> {
-                            label.setIsRemoved(false);
-                            label.setIsFaded(false);
-                        });
-            } else {
-                allLabels.stream()
-                        .filter(label -> label.getActualName().equals(name))
-                        .filter(label -> resultList.get(label.getActualName()))
-                        .filter(label -> !isInTopLabels(label.getActualName()))
-                        .findFirst()
-                        .ifPresent(label -> topLabels.add(new PickerLabel(label, this, true)));
-            }
-        } else {
-            topLabels.stream()
-                    .filter(label -> label.getActualName().equals(name))
-                    .findFirst()
-                    .ifPresent(label -> {
-                        if (issue.getLabels().contains(name)) {
-                            label.setIsRemoved(true);
-                        } else {
-                            topLabels.remove(label);
-                        }
-                    });
-        }
-    }
-
-    private boolean isInTopLabels(String name) {
-        // used to prevent duplicates in topLabels
-        return topLabels.stream()
-                .filter(label -> label.getActualName().equals(name))
-                .findAny()
-                .isPresent();
+    public LabelPickerUILogic() {
```

Comment:
Do we still need this class? Given that it is stateless and doesn't have much inside it. If you want to keep it, the methods can be static.